kolibri/core/auth/serializers.py

```diff
@@ -28,13 +30,63 @@
 
 
 class RoleListSerializer(serializers.ListSerializer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.invalid_items = []
+
+    def validate(self, attrs):
+
+        # Separate classroom-level coach roles from others
+        classroom_coach_items = []
+
+        # Other items is for items that don't need to be validated for
+        # whether or not they're enrolled in a class already
+        other_items = []
+
+        # Tracking the collections & users to query against for validating
+        # if they're members of the class already
+        assignable_coach_ids = []
+        class_collection_ids = []
+
+        for role_data in attrs:
+            collection = role_data["collection"]
+            kind = role_data["kind"]
+
+            # Only validate classroom-level coach roles
```

Comment:
I am a bit confused by this - why are we only preventing classroom assignable coaches from being assigned and enrolled in a class, but not putting similar restrictions on facility coaches and admins?